desc/magnetic_fields.py

```diff
@@ -1087,6 +1289,319 @@ def odefun(rpz, s):
     return r, z
 
 
+### Dommaschk potential utility functions ###
+
+# based off Representations for vacuum potentials in stellarators
+# https://doi.org/10.1016/0010-4655(86)90109-8
+
+# written with naive for loops initially and can jax-ify later
+
+true_fun = lambda m_n: 0.0  # used for returning 0 when conditionals evaluate to True
+
+
+@jit
+def gamma(n):
+    """Gamma function, only implemented for integers (equiv to factorial of (n-1))."""
+    return jnp.exp(gammaln(n))
+
+
+@jit
+def alpha(m, n):
+    """Alpha of eq 27, 1st ind comes from C_m_k, 2nd is the subscript of alpha."""
+    # modified for eqns 31 and 32
+
+    def false_fun(m_n):
+        m, n = m_n
+        return (-1) ** n / (gamma(m + n + 1) * gamma(n + 1) * 2.0 ** (2 * n + m))
+
+    def bool_fun(n):
+        return jnp.any(n < 0)
```

Comment:
do we need `jnp.any` in all of these? aren't m, n always scalars?

Reply:
I think you're right, I was just going trigger happy on the jnp statements when I converted it to JAX